var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -583,6 +583,8 @@ class Llvm(CMakePackage):
 
     # Github issue #4986
     patch('llvm_gcc7.patch', when='@4.0.0:4.0.1+lldb %gcc@7.0:')
+    # Backport from llvm master and add additional test.
```

Comment:
Please reference the upstream bug report for this: https://bugs.llvm.org/show_bug.cgi?id=39696 and also note the the upstream patch to resolve the bug is broken.  After that this pr should be good to go.